### PR TITLE
feat(ui): support multi selection in chat panel

### PR DESCRIPTION
**Pull request by liangfung**
### Screen record
https://jam.dev/c/85b756fd-4659-4ca6-a77b-b2df860cf2c1

### CODE REVIEW

File: clients/tabby-chat-panel/src/index.ts [hunk 1]
```diff
@@ -13,7 +13,10 @@ export interface FileContext {
   git_url: string
 }
 
-export type Context = FileContext
+export type Context = FileContext & {
+  // FIXME: The `isActiveSelection` field is used to differentiate whether it is the active selection or the multi selection added by command.
+  isActiveSelection?: boolean
```
Comment (liangfung) [inline clients/tabby-chat-panel/src/index.ts:18]: The `isActiveSelection` field is used to differentiate whether it is the active selection or the multi selection added by command

File: ee/tabby-ui/components/chat/chat.tsx [hunk 2]
```diff
@@ -303,8 +303,10 @@ function ChatRenderer(
     const userMessage = qaPairs[qaPairs.length - 1].user
     // FIXME(wwayne): The first context in relevantContext is currently sent as the code query.
     //                Review and update the logic to ensure the appropriate api attribute is used.
+    const relevantContextFromActiveSelection =
+      userMessage?.relevantContext?.find(o => o.isActiveSelection)
     const contextForCodeQuery =
-      userMessage?.selectContext || userMessage?.relevantContext?.[0]
+      userMessage?.selectContext || relevantContextFromActiveSelection
```
Comment (liangfung) [inline ee/tabby-ui/components/chat/chat.tsx:309]: The relevant context from the active selection will be passed into `code_query`.

**Review by icycodes — APPROVED** [review]

File: clients/tabby-chat-panel/src/index.ts (hunk 1 above)
Comment (wsxiaoys) [reply clients/tabby-chat-panel/src/index.ts:18]: I recommend adding a field in `ChatMessage`, e.g `activeContext` to represent context being selected

Comment (wsxiaoys) [reply clients/tabby-chat-panel/src/index.ts:18]: I recommend adding a new field in ChatMessage to represent such context, e.g `activeContext`

Comment (liangfung) [reply clients/tabby-chat-panel/src/index.ts]: Updated, added an `activeContext` field in `ChatMessage`.

**Pull request merged**